### PR TITLE
Montana Child Tax Credit

### PR DESCRIPTION
**Pull request by FlameO3O**
### Summary
:sparkles::page_facing_up::abacus:

This pull request adds a new feature to the policy engine: the Montana child tax credit. It creates new parameter, variable, and test files for the credit, and updates the changelog accordingly. The credit is based on the number and age of dependents, and the income and investment income of the tax unit.

> _`Montana` credit_
> _New feature for young children_
> _Spring brings tax relief_

### Walkthrough
*  Add a new feature: the Montana child tax credit ([link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
*  Create new parameter files for the credit amount, eligibility, and reduction rules ([link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-1e6d3d297aa1081c015b45259c2a7ecc8f36135b39b28401818d9d276e31a2d5R1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-4417e79ab4af3179a02a095aaec0ea8a040b10f0897f5b68477c0983aad90474R1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-3d4cd3f4a57b861bd17a7a24249107ed7a9f5931303f06083e995501cebdebbaR1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-48318e896e028c1c616fa99876e608aaa403e5239e4236a56131eb4d5515f57dR1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-eaf3abd8da41cc1c9932529282bab8147bd7ee18985465b137d0bb68fb96d763R1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-ba65ed7db9611143053944ca6c4a3afcc41556aac5a295128cba09f23ddda8b4R1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-ae2a752a17be53826a7c01c3187ceb097f9898196e8f6cab4d569981e398e8daR1-R10))
*  Define a new variable `mt_ctc` that calculates the credit based on the parameters and the tax unit data ([link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-4d150a80e0ce559fa6266b7bda12ad31349549f16a8d1170d07c266e4d0a2f96R1-R36))
*  Add a new test file `mt_ctc.yaml` that checks the output of the variable against expected values for different scenarios ([link](https://github.com/PolicyEngine/policyengine-us/pull/2675/files?diff=unified&w=0#diff-2190be8e8f76193d359deac5574a7c1c734ca7448ab9039536af38cb7d501e1eR1-R80))

### CODE REVIEW

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from policyengine_us.model_api import *
+
+
+class mt_ctc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "MT CTC"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc.py:7]:
```suggestion
    label = "Montana CTC"
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/child_age_eligibility.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+description: Child Age For Child Tax Credit Eligibility.
+values:
+  2023-01-01: 5
+metadata:
+  units: year
```
Comment (ShaylaWoo) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/child_age_eligibility.yaml:5]: `units: age`

Comment (MaxGhenis) [reply policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/child_age_eligibility.yaml:5]: Actually we do use year as the unit for age

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/base.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+description: Montana provides the following base amount per child under the child tax credit.
+values:
+  2023-01-01: 1_200
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/base.yaml:3]: need 2021 values + tax form reference

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/base.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+description: Montana provides the following base amount per child under the child tax credit.
+brackets:
+  - threshold:
+      2021-11-01: 0
+    amount:
+      2023-01-01: 1_200
+      2021-11-01: 3_600
+      2020-01-01: 2_000
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/base.yaml:8]: older values first:
2020-01-01:
2021-01-01:
2023-01-01:

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/base.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  threshold_unit: year
+  amount_unit: currency-USD
+  period: year
+  name: montana_child_tax_credit_base
+  label: "Montana child tax credit base"
+  reference:
+    - title: 2023 Montana Legislature, House Bill No. 268, Section 1, 7 (c)
+      href: https://leg.mt.gov/bills/2023/billhtml/HB0268.htm
+    - title: Federal Tax Credit, page 3-4
+      href: https://leg.mt.gov/content/Committees/Interim/2021-2022/Economic%20Affairs/Meetings/November%202021/Federal-Tax-Credits.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/base.yaml:36]:
```suggestion
      href: https://leg.mt.gov/content/Committees/Interim/2021-2022/Economic%20Affairs/Meetings/November%202021/Federal-Tax-Credits.pdf#page=3
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/child_age_eligibility.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+description: Child Age For Child Tax Credit Eligibility.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/child_age_eligibility.yaml:1]:
```suggestion
description: Montana provides the child tax credit for children at or below this age threshold.
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/child_age_eligibility.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+values:
+  2023-01-01: 5
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/child_age_eligibility.yaml:3]: need 2021 value

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/child_age_eligibility.yaml:3]: all yaml

**Review by PavelMakarchuk — APPROVED** [review]

**Comment by MaxGhenis** [comment]
i'll review after all outstanding comments are resolved and tests pass

File: policyengine_us/tests/policy/baseline/gov/states/mt/tax/income/credits/ctc/mt_ctc.yaml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        age: 3
+      dep2:
+        age: 5
+      dep3:
+        age: 6
+    tax_units:
+      tax_unit:
+        members: [head, dep1, dep2, dep3]
+        mt_ctc_eligible: false
+        adjusted_gross_income: 49_999
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/mt/tax/income/credits/ctc/mt_ctc.yaml:76]: test fails

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_threshold.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_threshold.yaml:1]: Move these to `eligibility/income_limit/{agi,investment}.yaml`

(I know they describe this as a threshold but limit is more consistent)

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_threshold.yaml [hunk 9]
```diff
@@ -0,0 +1,13 @@
+description: Montana provides the child tax credit for filers with income below this threshold.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_threshold.yaml:1]:
```suggestion
description: Montana provides the child tax credit for filers with federal adjusted gross income below this amount.
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_threshold.yaml [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+values:
+  2021-01-01: 0
+  2023-01-01: 56_000
+metadata:
+  units: currency-USD
+  period: year
+  label: Montana child tax credit income threshold
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_threshold.yaml:8]:
```suggestion
  label: Montana child tax credit income limit
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_threshold.yaml [hunk 11]
```diff
@@ -0,0 +1,13 @@
+description: Montana provides the child tax credit for filers with income below this threshold.
+values:
+  2021-01-01: 0
+  2023-01-01: 56_000
+metadata:
+  units: currency-USD
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_threshold.yaml:6]:
```suggestion
  unit: currency-USD
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/increment.yaml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+description: Montana reduces the child tax credit in the following increments.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/increment.yaml:1]:
```suggestion
description: Montana reduces the child tax credit for each of these increments that a filer's adjusted gross income exceeds the threshold.
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+    value_type = bool
+    entity = TaxUnit
+    label = "Eligible for the Montana Child Tax Credit"
+    definition_period = YEAR
+    reference = "https://leg.mt.gov/bills/2023/billpdf/HB0268.pdf"
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.mt.tax.income.credits.ctc
+        # CTC limited to filers with agi at or below $56,000
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py:14]:
```suggestion
        # CTC limited to filers with AGI at or below $56,000
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+        agi = tax_unit("adjusted_gross_income", period)
+        income_eligible = agi <= p.income_threshold
+        # CTC limited to filers with investment income below $10,300
+        investment_income_eligible = (
+            tax_unit("net_investment_income", period) < p.investment_threshold
+        )
+        return income_eligible & investment_income_eligible
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py:21]: Add 4(A): "proof of earned income" (`tax_unit_earned_income > 0`)

Let's also make this a bool parameter: `eligibility/earned_income_required.yaml`

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/base.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/base.yaml:1]: rename `amount.yaml`

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/agi.yaml [hunk 15]
```diff
@@ -0,0 +1,13 @@
+description: Montana provides the child tax credit for filers with federal adjusted gross income below this amount.
+values:
+  2021-01-01: 0
+  2023-01-01: 56_000
+metadata:
+  unit: currency-USD
+  period: year
+  label: Montana child tax credit investment income limit
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/agi.yaml:8]:
```suggestion
  label: Montana child tax credit income limit
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/earned_income_required.yaml [hunk 16]
```diff
@@ -0,0 +1,13 @@
+description: Montana provides the child tax credit with earned income above this threshold.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/earned_income_required.yaml:1]:
```suggestion
description: Montana limits its child tax credit to filers with earned income above this threshold.
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/earned_income_required.yaml [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+values:
+  2021-01-01: 0
+metadata:
+  units: currency-USD
+  period: year
+  label: Montana child tax credit earned income limit
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/earned_income_required.yaml:7]:
```suggestion
  label: Montana child tax credit earned income threshold
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/investment.yaml [hunk 18]
```diff
@@ -0,0 +1,15 @@
+description: Montana provides the child tax credit for filers with investment income below this threshold.
+values:
+  2021-01-01: 0
+  2023-01-01: 10_300
+metadata:
+  units: currency-USD
+  period: year
+  label: Montana child tax credit investment income threshold
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/investment.yaml:8]:
```suggestion
  label: Montana child tax credit investment income limit
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/investment.yaml [hunk 19]
```diff
@@ -0,0 +1,15 @@
+description: Montana provides the child tax credit for filers with investment income below this threshold.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/investment.yaml:1]:
```suggestion
description: Montana limits its child tax credit to filers with investment income below this amount.
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/rate.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/rate.yaml:1]: rename `amount`. `rate` typically implies percentage in our system

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Comment by ShaylaWoo** [comment]
We came across two problems when fixing the changes:
1. There exists policyenige_us/parameters/gov/states/mt/tax/income/credits/ctc/amount.yaml. Is it possible to get the same name but a different folder?
2. Lack of law resource for verifying the mt_ctc_eligible.py's alignment.

**Comment by PavelMakarchuk** [comment]
Duplicate

**Pull request closed without merging**